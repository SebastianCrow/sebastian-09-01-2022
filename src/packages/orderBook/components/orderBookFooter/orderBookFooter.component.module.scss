@import '../../../../styles/variables';

.toggleFeedContainer {
  display: flex;
  justify-content: center;
  padding: 2 * $spacing;
}
