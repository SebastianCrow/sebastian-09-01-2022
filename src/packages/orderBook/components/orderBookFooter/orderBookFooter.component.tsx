import React, { FunctionComponent, useCallback } from 'react';
import styles from './orderBookFooter.component.module.scss';
import { Button } from '../../../ui/button/button.component';
import { useSelectOrderBookState } from '../../hooks/useSelectOrderBookState.hook';
import { Product } from '../../services/network/orderBookNetwork.types';
import { FormattedMessage } from 'react-intl';

interface OrderBookFooterProps {
  observeProduct: (product: Product) => void;
}

export const OrderBookFooter: FunctionComponent<OrderBookFooterProps> = ({
  observeProduct,
}) => {
  const { product, connectionStatus } = useSelectOrderBookState();

  const toggleFeed = useCallback(() => {
    observeProduct(product === 'Bitcoin' ? 'Ethereum' : 'Bitcoin');
  }, [observeProduct, product]);

  return (
    <div className={styles.toggleFeedContainer}>
      <Button onClick={toggleFeed} disabled={connectionStatus === 'error'}>
        <FormattedMessage id="Toggle Feed" />
      </Button>
    </div>
  );
};
