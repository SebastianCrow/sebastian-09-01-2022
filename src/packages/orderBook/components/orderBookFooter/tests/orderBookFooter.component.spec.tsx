import React from 'react';
import { act, ReactTestRenderer } from 'react-test-renderer';
import { createRendererWithOrderBookStore } from '../../../tests/utils/createRendererWithOrderBookStore.util';
import { findByDataTestId } from '../../../../../tests/utils/findByDataTestId.util';
import { OrderBookFooter } from '../orderBookFooter.component';
import { renderWithOrderBookStore } from '../../../tests/utils/renderWithOrderBookStore.util';
import { fireEvent } from '@testing-library/react';

describe('orderBookFooter.component', () => {
  let renderer: ReactTestRenderer;

  test('renders', () => {
    act(() => {
      renderer = createRendererWithOrderBookStore(
        <OrderBookFooter observeProduct={jest.fn()} />
      );
    });

    expect(renderer.toJSON()).toMatchSnapshot();
    expect(findByDataTestId(renderer.root, 'toggle-feed-button')).toBeTruthy();
  });

  test('toggle feed button triggers feed observation', () => {
    const observeProduct = jest.fn();
    // TODO: Is act() required?
    const { getByTestId } = renderWithOrderBookStore(
      <OrderBookFooter observeProduct={observeProduct} />
    );
    fireEvent(
      getByTestId('toggle-feed-button'),
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      })
    );
    expect(observeProduct).toHaveBeenCalledTimes(1);
  });
});
