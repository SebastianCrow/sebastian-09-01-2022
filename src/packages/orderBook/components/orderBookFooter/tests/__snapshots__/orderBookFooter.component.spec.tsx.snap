// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderBookFooter.component renders 1`] = `
<div
  className="toggleFeedContainer"
>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root primary css-sghohy-MuiButtonBase-root-MuiButton-root"
    data-testid="toggle-feed-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Toggle Feed
  </button>
</div>
`;
